rest_framework/utils/encoders.py

```diff
@@ -41,7 +41,10 @@ def default(self, obj):
                 representation = representation[:12]
             return representation
         elif isinstance(obj, datetime.timedelta):
-            return six.text_type(obj.total_seconds())
+            if hasattr(obj, 'total_seconds'):
+                return six.text_type(obj.total_seconds())
+            else:  # Python 2.6 compatibility
+                return six.text_type((obj.microseconds + (obj.seconds + obj.days * 24 * 3600) * 10**6) / 10**6)
```

Comment:
We isolate any version branching code into `compat.py`.
Have a function named `total_seconds(timedelta)` that provides this in `compat.py` and then call that from here.
Otherwise seems fine.